lib/ansible/modules/cloud/amazon/route53_health_check.py

```diff
@@ -185,8 +185,16 @@ def health_check_diff(a, b):
         return {}
     diff = {}
     for key in set(a.keys()) | set(b.keys()):
-        if a.get(key) != b.get(key):
-            diff[key] = b.get(key)
+        a_val = a.get(key)
+        b_val = b.get(key)
+        a_type = type(a_val)
+        b_type = type(b_val)
+        if (a_type is int and b_type is unicode):
```

Comment:
Instead of unicode, it's safest to compare against text_type since Py2 and Py3 have different types. 

Alternatively, it seems it'd be sufficient to do: `if int(a.get(key)) != int(b.get(key))` since running `int()` on an integer is a no-op.